### ISSUE TITLE
Importance of probability calibration for classifiers

### ISSUE THREAD

**Issue by dvas0004** [dvas0004/NerdNotes#74]
Classifiers in this note refer to:

- algorithmic such as SVM or Random Forests. In this case the scikit `predict_proba` function is sometimes used as a measure of confidence
- Neural Networks such as NNs ending with sigmoid layer mapping to class. In this case the output of the sigmoid function is sometimes used as a measure of confidence

PS, in anomaly detection if the output of the confidence is sufficiently low, the datapoint can be considered an anomaly.

Both types of classifiers however sometimes skew the results of their confidence measures [(1)](https://scikit-learn.org/stable/auto_examples/calibration/plot_calibration.html#sphx-glr-auto-examples-calibration-plot-calibration-py)[(2)](https://stats.stackexchange.com/questions/309642/why-is-softmax-output-not-a-good-uncertainty-measure-for-deep-learning-models?rq=1)

In order to resolve this, probability calibration is sometimes necessary. Basically the output of each classifier's confidence is passed through a regressor that has been trained on the predicted vs actual confidence. In other words:

`1. The classifier is trained as usual`
`2. For each output class, create a regressor` ([isotonic](https://scikit-learn.org/stable/auto_examples/plot_isotonic_regression.html#sphx-glr-auto-examples-plot-isotonic-regression-py) or normal)
`3. For every training sample, pass it through the model, and for each output class record the predicted probability and the actual probability (which is usually 0, or 1)`
`4. Train the regressors created in step 2 on the data collected in step 3`

**Comment by dvas0004** [comment]
http://fastml.com/classifier-calibration-with-platts-scaling-and-isotonic-regression/
